app/components/main-content/PrelaunchSignUpForm.tsx

```diff
@@ -2,44 +2,39 @@
 
 import { TextInput, Button, Box } from '@mantine/core';
 import { useForm } from '@mantine/form';
-import { useEffect, Dispatch, SetStateAction } from 'react';
+import { useEffect, Dispatch, SetStateAction, useMemo } from 'react';
 import { FlowState } from '../../early-access/page'
 import { getRedirectResult } from 'firebase/auth';
 import {
   auth,
   signUpWithGitHub,
   signUpWithGoogle,
   createUserDoc,
-} from '@/app/config/firebase'; 
+} from '@/app/config/firebase';
 import { doc, getDoc, setDoc, serverTimestamp } from 'firebase/firestore';
 import { db } from '@/app/config/firebase';
+import { generateInitialValues, generateValidationRules } from '@/app/utils/formInitialization';
+import textInputConfig from '../../config/signupForm';
+import { FormValues } from '@/app/types/signupForm';
 
 interface PrelaunchSignUpFormProps {
   setFlowState: Dispatch<SetStateAction<FlowState>>;
 }
 
-interface FormValues {
-  name: string;
-  email: string;
-}
-
 const PrelaunchSignUpForm: React.FC<PrelaunchSignUpFormProps> = ({ setFlowState }) => {
 
-  const form = useForm({
-    initialValues: {
-      name: '',
-      email: '',
-    },
```

Comment:
Line 26 & 27 initialize `initialValues` and `validationRules` using the `useMemo` hook to optimize the performance of the `PrelaunchSignUpForm` component. By passing an empty dependency array (`[]`) to `useMemo`, it ensures that `generateInitialValues` and `generateValidationRules` functions are only invoked once during the initial render.